src/components/bookAdd.js

```diff
@@ -1,11 +1,48 @@
-import React from 'react';
-
-const BookAdd = () => (
-  <form>
-    <input type="text" name="title" />
-    <input type="text" name="author" />
-    <button type="submit">Add Book</button>
-  </form>
-);
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
+
+const BookAdd = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('');
+
+  const handleAddBook = (e) => {
+    e.preventDefault();
+    if (!title.trim() || !author.trim() || !category.trim()) return;
+    dispatch(addBook({ title, author, category }));
+    setTitle('');
+    setAuthor('');
+    setCategory('');
+  };
+
+  return (
+    <form>
+      <input
+        type="text"
+        name="title"
+        value={title}
+        onChange={(e) => setTitle(e.target.value)}
+        placeholder="Title"
+      />
+      <input
+        type="text"
+        name="author"
+        value={author}
+        onChange={(e) => setAuthor(e.target.value)}
+        placeholder="Author"
+      />
+      <input
+        type="text"
+        name="category"
+        value={category}
+        onChange={(e) => setCategory(e.target.value)}
+        placeholder="Category"
+      />
+      <button type="submit" onClick={handleAddBook} className="add-btn">Add Book</button>
```

Comment:
- [ ] This is a good start, but it seems that one of the project requirements has been overlooked. Specifically, you should have a `<Button>` component for adding a book to the books array. Please replace the existing `<button>` element with this component, just as you have done for removing a book in the Book component. This will help to ensure consistency and maintainability across the application. Thank you for your attention to this matter.